base-image/Dockerfile

```diff
@@ -96,13 +97,13 @@ ONBUILD RUN echo "Checking for 'conda-linux-64.lock' or 'environment.yml'..." \
         ; [ -d binder ] && cd binder \
         ; [ -d .binder ] && cd .binder \
         ; if test -f "conda-linux-64.lock" ; then \
-        conda create --name ${CONDA_ENV} --file conda-linux-64.lock \
+        mamba create --name ${CONDA_ENV} --file conda-linux-64.lock \
         ; elif test -f "environment.yml" ; then \
-        conda env create --name ${CONDA_ENV} -f environment.yml  \
+        mamba env create --name ${CONDA_ENV} -f environment.yml  \
```

Comment:
seems like this is the only case where there will be a difference, since using a .lock file bypasses solving the environment. if we do incorporate this, we should change the CI condalock steps to use mamba as well (https://github.com/conda-incubator/conda-lock/pull/32).

Reply:
i'm guessing mamba speed improvement is most noticeable with many packages/pins in environment.yml. currently solving for the 3 environments in this repository only takes about 7 minutes total https://github.com/pangeo-data/pangeo-docker-images/runs/1184620958?check_suite_focus=true

Reply:
I suspect, though I don't know this for sure, that mamba and conda will produce identical environments for this step. My hypothesis is also that mamba will be a bit faster here but that the most noticeable improvements will come when the environment solving steps.